src/utils/AuthService.ts

```diff
@@ -109,113 +112,50 @@ function initializeAuthenticationDev(
             handleSigninSilent(dispatch, userManager);
         }
     }
-    return Promise.resolve(userManager);
+    return userManager;
 }
 
 const accessTokenExpiringNotificationTime = 60; // seconds
 
-function initializeAuthenticationProd(
+export async function initializeAuthenticationProd(
     dispatch: Dispatch<unknown>,
     isSilentRenew: boolean,
-    idpSettings: Promise<Response>,
+    idpSettingsFetcher: IdpSettingsGetter,
     validateUser: UserValidationFunc,
-    authorizationCodeFlowEnabled: boolean,
     isSigninCallback: boolean
 ) {
-    return idpSettings
-        .then((r) => r.json())
```

Comment:
Actually I'm not really sure this brings any value... What's better than before doing it in the apps and not here ?

Reply:
duplicated step in all fronts, and the same than why the api fetchers functions in apps do the `.json()` themselves

Reply:
is it okay to do like that to keep compatibility for gridcapa?